three go tests fail on latest commit of master (0d9f544bb94a)

### My Setup

* commit 0d9f544bb94aac29c185968483459ef74d6deb5d
* Go version: `go version go1.7.1 darwin/amd64`
* `uname -a` == `Darwin Thomass-MBP.libcore.so 14.5.0 Darwin Kernel Version 14.5.0: Sun Sep 25 22:07:15 PDT 2016; root:xnu-2782.50.9~1/RELEASE_X86_64 x86_64`

### Recipe

```
go test -v -quick.seed=21691
```

# What I Did

I ran this script, which is equivalent to running `go test -v`, but it only runs certain tests.

```
#!/bin/sh
go test -v -run "(TestBucket_Stats_Large|TestBucket_Put_Multiple|TestBucket_Delete_Quick|TestCursor_QuickCheck$|TestCursor_QuickCheck_Reverse|TestOpen_Size|TestDB_Open_InitialMmapSize|TestDB_Close_PendingTx_RW|TestDB_Close_PendingTx_RO|TestSimulate_10000op_1p|TestSimulate_10000op_10p|TestSimulate_10000op_100p|TestSimulate_10000op_1000p)"
```

# What I Expected

I expected all tests to pass.

# What Actually Happened

`TestBucket_Put_Multiple`, `TestCursor_QuickCheck`, and `TestCursor_QuickCheck_Reverse` failed. Here's the output of the above script.

```
seed: 21691
quick settings: count=5, items=1000, ksize=1024, vsize=1024
=== RUN   TestBucket_Stats_Large
--- PASS: TestBucket_Stats_Large (0.34s)
=== RUN   TestBucket_Put_Multiple
db copied to:  /var/folders/f7/1n6mcqws1bj1tt2kbhjxtc7r0000gn/T/bolt-680967459
--- FAIL: TestBucket_Put_Multiple (0.06s)
	bucket_test.go:1687: exp=2080242161f488f5d161d8cd8eb3846bf3b0fabb03de33214c1a962e973d2383851ffa848aa415431136d6931316aa60dc8c64ebe606376391b193782e3f76ba8090a2e254eb51744d124d4112a718fa675c80f31f4baae1ccfed975b16aca47cd884fe9cb90da93f25a41840092f0df848e60477f8c41deaef7291a3f9943dd; got=5be5c99a5998604c7d48eab0c15217adbe6c2cdece5528c86fab14998625914ddeb7674aa62daed8d892f1a1b03d2559720479a022da4b368161ba0aa7b80e917116013164a43478c6bdae3925188508
=== RUN   TestBucket_Delete_Quick
--- PASS: TestBucket_Delete_Quick (0.68s)
=== RUN   TestCursor_QuickCheck
--- FAIL: TestCursor_QuickCheck (0.05s)
	cursor_test.go:547: unexpected key: [99 99 169 233 212 23 11 179 31 66 218 1 191 146 21 30 88 152 157 212 222 237 45 243 91 103 39 115 103 155 113 186 215 33 36 201 238 50 91 230 121 119 84 217 54 65 112 174 87 74 154 238 11 156 179 226 34 76 239 42 6 137 136 25 35 80 207 178 76 86 95 69 237 58 7 141 151 185 147 179 113 198 0 136 184 11 217 90 17 113 123 2 80 154 206 107 251 54 63 228 59 230 130 173 31 52 45 51 107 196 162 239 167 96 8 28 243 19 130 108 183 54 239 241 36 154 158 202 249 169 43 99 222 61 88 35 185 77 147 82 66 95 108 127 68 73 195 44 213 173 135 156 123 37 3 59 230 111 111 32 154 207 203 142 185 254 142 25 249 204 108 211 247 3 26 55 214 174 46 116 71 246 248 193 106 88 83 196 225 108 237 232 125 27 213 152 132 99 148 148 105 182 56 178 21 241 138 34 222 164 250 99 229 22 106 8 111 206 131 145 214 200 26 246 91 8 53 24 183 249 111 219 63 13 246 230 102 197 250 151 76 170 125 47 141 196 119 247 176 146 219 32 79 192 75 189 210 206 213 85 62 181 163 135 177 23 30 84 54 148 158 66 153 177 32 42 125 227 73 101 204 173 60 45 51 149 243 70 143 238 43 182 57 40 55 240 122 136 68 29 92 41 221 145 112 65 143 198 247 205 122 22 214 0 56 239 43 123 155 12 233 232 103 174 228 137 245 21 139 225 180 244 116 189 102 174 72 155 78 203 47 207 223 0 150 156 58 59 40 8 148 28 133 37 173 17 165 202 179 244 57 75 103 241 4 240 45 198 181 64 94 212 141 174 51 115 214 31 72 136 100 22 126 206 0 114 5 148 178 218 104 32 70 160 175 25 20 1 239 19 159 2 244 215 250 158 97 108 198 240 99 40 224 99 224 197 197 206 110 155 198 25 61 158 153 167 95 105 77 153 168 53 232 240 127 217 4 238 213 254 82 16 228 240 41 10 76 249 106 230 205 86 77 26 219 188 202 59 66 86 30 84 71 140 68 241 36 43 127 108 123 67 213 22 127 72 146 203 243 158 10 228 23 177 224 83 236 147 197 223 95 194 175 114 71 167 127 183 126 85 35 47 238 24 14 253 14 92 68 199 167 105 191 153 213 89 108 132 25 36 72 224 141 194 246 203 184 84 68 67 200 136 12 66 65 145 249 74 51 71 219 159 26 174 229 14 157 68 218 166 214 130 119 202 204 111 163 233 233 140 103 247 185 49 119 39 75 200 254 165 210 192 15 66 222 81 25 38 45 178 249 80 47 108 59 123 140 172 113 214 109 67 19 224 22 103 240 73 136 210 142 28 26 28 73 56 114 74 146 242 133 162 99 185 131 126 67 240 30 121 205 62 39 237 180 36 249 141 127 210 190 16 219 224 62 206 13 244 171 133 232 217 137 78 135 104 21 45 235 170 249 254 74 212 197 85 153 154 98 144 50 229 62 11 109 181 203 54 163 156 154 147 99 111 181 193 46 217 99 79 39 77 254 123 97 244 23 82 13 198 214 40 38 161 116 78 16 149 169 47 4 155 133 165 107 167 213 187 192 134 156 245 178 128 195 80 53 204 250 116 90 242 3 193 106 80 135 148 115 59 173 102 240 189 31 66 182 239 86 131 105 216 143 180 144 105 105 66 39 126 202 57 232 77 145 237 95 232 111 232 184 200 154 198 178 32 18 98 119 203 198 184 85 49 73 232 228 122 82 6 222 221 254 253 25 109 142 157 81 229 197 132 125 156 165 111 48 101 53 181 111 26 219 204 235 32 215 171 199 177 124 72 58 102 65 121 211 224 114 132 91 197 46 70 82 122 92 163 183 62 96 196 149 157 17 113 170 103 4 11 86 117 140 58 6 137 217 43 251 151 212 59 247 113 200 223 127]
=== RUN   TestCursor_QuickCheck_Reverse
--- FAIL: TestCursor_QuickCheck_Reverse (0.05s)
	cursor_test.go:606: unexpected value: [91 229 201 154 89 152 96 76 125 72 234 176 193 82 23 173 190 108 44 222 206 85 40 200 111 171 20 153 134 37 145 77 222 183 103 74 166 45 174 216 216 146 241 161 176 61 37 89 114 4 121 160 34 218 75 54 129 97 186 10 167 184 14 145 113 22 1 49 100 164 52 120 198 189 174 57 37 24 133 8]
=== RUN   TestOpen_Size
--- PASS: TestOpen_Size (0.21s)
=== RUN   TestOpen_Size_Large
--- PASS: TestOpen_Size_Large (18.89s)
=== RUN   TestDB_Open_InitialMmapSize
--- PASS: TestDB_Open_InitialMmapSize (0.38s)
=== RUN   TestDB_Close_PendingTx_RW
--- PASS: TestDB_Close_PendingTx_RW (0.21s)
=== RUN   TestDB_Close_PendingTx_RO
--- PASS: TestDB_Close_PendingTx_RO (0.20s)
=== RUN   TestSimulate_10000op_1p
--- PASS: TestSimulate_10000op_1p (2.20s)
=== RUN   TestSimulate_10000op_10p
--- PASS: TestSimulate_10000op_10p (2.05s)
=== RUN   TestSimulate_10000op_100p
--- PASS: TestSimulate_10000op_100p (2.53s)
=== RUN   TestSimulate_10000op_1000p
--- PASS: TestSimulate_10000op_1000p (1.98s)
FAIL
exit status 1
FAIL	github.com/boltdb/bolt	29.943s
```

# Background

I'm developing a program that exposes statistically significant performances differences in programs. It can parse `go test -v` output to collect data. The program has to collect a decent amount of data to find anything statistically significant, so `go test -v` (or in this case, a provided script) has to be run a lot. **The above script had run 26 times successfully before it failed.** I don't know all that much about boltdb, but I suspect this might be able to be replicated if the seed in the above output is used.

--- a/quick_test.go
+++ b/quick_test.go
@@ -50,9 +50,17 @@ func (t testdata) Less(i, j int) bool { return bytes.Compare(t[i].Key, t[j].Key)
 func (t testdata) Generate(rand *rand.Rand, size int) reflect.Value {
 	n := rand.Intn(qmaxitems-1) + 1
 	items := make(testdata, n)
+	used := make(map[string]bool)
 	for i := 0; i < n; i++ {
 		item := &items[i]
-		item.Key = randByteSlice(rand, 1, qmaxksize)
+		// Ensure that keys are unique by looping until we find one that we have not already used.
+		for {
+			item.Key = randByteSlice(rand, 1, qmaxksize)
+			if !used[string(item.Key)] {
+				used[string(item.Key)] = true
+				break
+			}
+		}
 		item.Value = randByteSlice(rand, 0, qmaxvsize)
 	}
 	return reflect.ValueOf(items)

